scene/resources/3d/navigation_mesh_source_geometry_data_3d.cpp

```diff
@@ -38,6 +38,16 @@ void NavigationMeshSourceGeometryData3D::set_indices(const Vector<int> &p_indice
 	indices = p_indices;
 }
 
+void NavigationMeshSourceGeometryData3D::append_vertices(const Vector<float> &p_vertices) {
+	RWLockWrite write_lock(geometry_rwlock);
+	vertices.append_array(p_vertices);
+}
+
+void NavigationMeshSourceGeometryData3D::append_indices(const Vector<int> &p_indices) {
+	RWLockWrite write_lock(geometry_rwlock);
+	indices.append_array(p_indices);
+}
+
```

Comment:
I think in many cases, the users will use those functions with array pairs indexed from 0. In such case, there will be some boilerplate code required on the client side to offset the arrays so that they work as expected after adding.

The above problem is very similar to what I've done here: https://github.com/lampe-games/godot-open-rts/commit/487e995ac47811627e9a0324906111a09990230d#diff-0291eaa88b59eec77f3d1546e6404bea1391d58eec1ee646874dded17f0bfe16R68-R69

So, IMO we should add extra `bool` parameter to allow user to control the append behavior on whether it should offset indices or not. Based on the value, the implementation would be as above or as here: https://github.com/godotengine/godot/pull/88221/files#diff-12521617192afd65ea274a9138a86fe63a3cdf4983caa2ca24e95ebd6541d8acR168-R177

Reply:
The more I think about it this should not be two functions and just one append_arrays() function.

With two functions it creates the same sync problems as with the NavigationMesh that has dedicated functions to set vertices and polygons. Without adding a commit() function you never know when the user actually has finished the update.

Reply:
Yeah, perhaps that would be better. In such case the indices offset should be aligned under the hood by default and without any booleans to control it. After all, if some user wants very low level access one can set indices directly.